sources/main.c

```diff
@@ -43,14 +42,19 @@ int	main(int argc, const char **argv)
 {
 	t_param	*param;
 
-	(void)argv;
 	param = param_init();
 	if (!param)
 		return (1);
-	if (argc == 1)
+	if (opts_fill(argc - 1, &argv[1], param))
+		return (finalize(param, 0, 0, 2));
+	pre(param);
+	if (param->opts.interactive)
 		interactive(param);
-	else
-		noninteractive(param, argc - 1, &argv[1]);
+	else if (param->cur_command)
```

Comment:
i dont see, where is` param->cur_command` assigned?
only one place before it happen is `interactive()`, but if `interactive` executed `else if(param->cur_command)` will not.